cmd/skopeo/proxy.go

```diff
@@ -614,6 +621,51 @@ func (h *proxyHandler) GetBlob(args []interface{}) (replyBuf, error) {
 	return ret, nil
 }
 
+func (h *proxyHandler) LayerInfoJSON(args []interface{}) (replyBuf, error) {
+	h.lock.Lock()
+	defer h.lock.Unlock()
+
+	var ret replyBuf
+
+	if h.sysctx == nil {
+		return ret, fmt.Errorf("client error: must invoke Initialize")
+	}
+
+	if len(args) != 1 {
+		return ret, fmt.Errorf("found %d args, expecting (imgid)", len(args))
+	}
+
+	imgref, err := h.parseImageFromID(args[0])
+	if err != nil {
+		return ret, err
+	}
+
+	ctx := context.TODO()
+
+	err = h.cacheTargetManifest(imgref)
+	if err != nil {
+		return ret, err
+	}
+	img := imgref.cachedimg
+
+	layerInfos, err := img.LayerInfosForCopy(ctx)
+	if err != nil {
+		return ret, err
+	}
+
+	if layerInfos == nil {
+		layerInfos = img.LayerInfos()
+	}
+
+	var custBlobArr []blobInfo
+	for _, blob := range layerInfos {
+		custBlobArr = append(custBlobArr, blobInfo{blob.Digest, blob.Size, blob.MediaType})
+	}
```

Comment:
`…blob := … layer…`

That sadly exists in the `c/image` type names, for ~historical reasons, but this code is specifically concerned about _layers_ (as they exist in c/storage; and as opposed to generic blobs in OCI artifacts, which can‘t be in c/storage). So maybe name this array, the iteration variable, and the `blobInfo` type to specifically refer to layers, not blobs.